### PR TITLE
Add CSS to default HTML template

### PR DESCRIPTION
**Pull request by mb21**
Two things that inspired this:

- questions like [this one on StackOverflow](https://stackoverflow.com/questions/63293929/pandoc-change-font-family-to-sans-while-converting-from-markdown-to-html)
- I've recently updated and streamlined the CSS I use to generate simple HTML files (or publish to my blog), and was surprised how minimal it can be, while still looking pretty good (IMHO). The idea is to rely on browser defaults as much as possible (which are fairly consistent nowadays), so you don't have to override/normalize a lot.

Maybe this has been discussed before, but what about including something like this in the default HTML template?

~~I also quickly looked into moving this into the `styles.html` partial, but there we don't have access to all those variables I assume?~~

TODO:

- [x] update the MANUAL, only new variable names are `fontcolor`, `backgroundcolor` (and `style`), although I suppose we could add a few more... considering how many we have for the LaTeX template... to allow even more customizations out of the box.
- [x] add a variable to disable outputting that whole CSS block? Perhaps we could also set that variable automatically when `--css` was used? But I cannot imagine any scenario where you'd use `--standalone`, but wouldn't want any CSS whatsoever.

### CODE REVIEW

File: data/templates/default.html5 [hunk 1]
```diff
@@ -15,6 +15,113 @@ $if(keywords)$
...
+      color: black;
+    }
+    nav a:not(:hover) {
+      text-decoration: none;
+    }
+    #TOC > ul {
+      margin-top: 1em;
+    }
+    #TOC:before {
+      content: 'Contents';
```
Comment (mb21) [inline data/templates/default.html5:120]: TODO: read out [our translation](https://github.com/jgm/pandoc/blob/master/data/translations/en.yaml#L6) into a variable and replace it here.

**Comment by jgm** [comment]
I'd have to think more about this, it's an interesting idea.

Note: the styles partial should have access to the same variables.

File: data/templates/default.html5 [hunk 2]
```diff
@@ -16,6 +16,7 @@ $endif$
   <title>$if(title-prefix)$$title-prefix$ – $endif$$pagetitle$</title>
   <style>
     $styles.html()$
+    $style$
```
Comment (jgm) [inline data/templates/default.html5:19]: IS this change intended?

Comment (mb21) [reply data/templates/default.html5:19]: yes, see [the MANUAL diff](https://github.com/jgm/pandoc/pull/6601/files#diff-1187c774ee83820d49b1f14f64e3aa4dR2181)

open to discuss all of it :-)

File: MANUAL.txt [hunk 3]
```diff
@@ -2142,6 +2142,56 @@ ODT or pptx.
...
 [Language subtag lookup]: https://r12a.github.io/app-subtags/
 
+### Variables for HTML
+
+`document-css`
+:   Enables inclusion of most of the CSS in the `styles.html`
+    [partial](#partials) (have a look with
+    `pandoc --print-default-data-file=templates/styles.html`).
+    This variable is set to `true` by default, unless you use
+    [`--css`](#option--css).
```
Comment (jgm) [inline MANUAL.txt:2152]: It occurs to me that this would be a breaking change -- currently, the bits of styling inserted by pandoc through styles partial are included even if you specify `--css`, though you can override them using `--css`.  Is there a reason not to keep this behavior?

Comment (mb21) [reply MANUAL.txt:2152]: The bits that were in the partial already before this change are not part of the `$if(document-css)$` block.. so they'll still always be included.

File: data/templates/default.html5 (hunk 2 above)
Comment (mb21) [reply data/templates/default.html5:19]: ah, but of course this doesn't work as intended, as it's interpreted as markdown... I'll remove it again. The README read:
> `style` can be used to provide additional CSS,
> potentially overriding the default. For example:
> 
> ```
> ---
> title: my doc
> style: |
>   blockquote {
>     border-left: none;
>   }
> ---
> ```

**Comment by jgm** [comment]
I decided to merge this; comments were favorable.  It was mentioned that this might give rise to endless requests for tweaks in the styling -- that is a danger, but we'll see!
